### PR TITLE
fix(core): fix drop partition SQL not removing the expected partition data and in some rare circumstances leading to a data loss 

### PR DESCRIPTION
**Pull request by mtopolnik**
As we're investigating an issue where dropping a partition ends up in a suspended, non-resumable table, this PR adds the DROP PARTITION operation to the existing battery of operations generated by the Fuzz test transaction generator.

It also fixes a bug in `DROP PARTITION WHERE ts < limit`, although probably not the one that causes a suspended table.

It also fixes another bug with incorrect column top managing when a column is added at the moment a split partition is the latest one, and then the partition is dropped.

### CODE REVIEW

File: core/src/test/java/io/questdb/test/AsyncDropPartitionTest.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            }
+            );
+            drainWalQueue.start();
+
+            // horse #2
+            TestUtils.await(barrier);
+            Thread.sleep(1);
+            System.out.println("Close reader");
+            r.close();
+//            System.out.println("Purge partition");
```
Comment (ideoma) [inline core/src/test/java/io/questdb/test/AsyncDropPartitionTest.java:87]: debris

File: core/src/test/java/io/questdb/test/griffin/wal/WalWriterFuzzTest.java [hunk 2]
```diff
@@ -120,33 +121,43 @@ public void testInOrderSmallTxns() throws Exception {
...
-        setFuzzProbabilities(0, 0.2, 0.1, 0, 0, 0, 0, 1.0, 0.01, 0.01, 0.0, 0);
+        setFuzzProbabilities(0, 0.2, 0.1, 0, 0, 0, 0, 0, 1.0, 0.01, 0.0, 0.01, 0.0);
         setFuzzCounts(rnd.nextBoolean(), rnd.nextInt(10_000_000),
                 rnd.nextInt(1500), 20, 10, 200, 0, 1
         );
         runFuzz(rnd);
     }
 
+    @Test
+    @Ignore("drop partition operation must be fixed")
```
Comment (ideoma) [inline core/src/test/java/io/questdb/test/griffin/wal/WalWriterFuzzTest.java:132]: Remove ignore and seeds

**Comment by glasstiger** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 43 / 115 (37.39%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/ColumnVersionReader.java|0|23|00.00%|
|:large_blue_circle:|io/questdb/cairo/TxReader.java|2|50|04.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompilerImpl.java|15|16|93.75%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnVersionWriter.java|20|20|100.00%|

**Comment by glasstiger** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 108 / 115 (93.91%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TxReader.java|43|50|86.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnVersionReader.java|23|23|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompilerImpl.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnVersionWriter.java|20|20|100.00%|

**Review by ideoma — DISMISSED** [review]

**Review by ideoma — APPROVED** [review]

**Comment by glasstiger** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 108 / 115 (93.91%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TxReader.java|43|50|86.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnVersionReader.java|23|23|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompilerImpl.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnVersionWriter.java|20|20|100.00%|

**Review by ideoma — APPROVED** [review]

**Pull request merged**